src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs

```diff
@@ -582,6 +579,17 @@ protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessa
                 }
             }
 
+            Uri? requestUri = request.RequestUri;
+            if (requestUri is null || !requestUri.IsAbsoluteUri)
```

Comment:
It seems like these are still things that should be checked regardless of handler.

Reply:
Though I suppose it's probably also good to check them here, in case we are being used without HttpClient...

Reply:
We ensure that the Uri is not null and IsAbsolute in [HttpClient](https://github.com/dotnet/runtime/blob/ca00588105feae898bc73b6d24bf9600d41b39db/src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs#L744), but we didn't do that in the handler directly.
I will add the check to WinHttpHandler as well.

Do we care to have the IsAbsolute check in Browser as well? I don't know if passing a relative Uri to it works / does anything useful.

Reply:
> Do we care to have the IsAbsolute check in Browser as well?

@lewing?

Reply:
The browser does support relative URIs. It could even support `null` URIs if they were treated like an empty string but they currently aren't. Relative URIs are resolved against `location.href`:
https://github.com/dotnet/runtime/blob/97898a4c9292d91a8e4f4b6add67f66f66f661b3/src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs#L244-L246

Reply:
To clarify, when using `HttpClient`, relative Uris will not be allowed through.

Where a user can get to calling `BrowsersHttpHandler` with a relative Uri is if they create the handler themselves and use an invoker instead of client:
```c#
using var invoker = new HttpMessageInvoker(new HttpClientHandler());
await invoker.SendAsync("relative Uri");
```

So if that does something meaningful, we should just let the browser decide what to do?


Side note:
`BrowsersHttpHandler` should most likely be calling
```c#
requestUri.IsAbsolute ? requestUri.AbsoluteUri : requestUri.ToString()
```
instead of just `ToString()` on the Uri.